addons/sale_expense/tests/test_sale_expense.py

```diff
@@ -97,3 +97,57 @@ def test_sale_expense(self):
         # both expenses should be invoiced
         inv = so._create_invoices()
         self.assertEqual(inv.amount_untaxed, 621.5 + (prod_exp_2.list_price * 100.0), 'Sale Expense: invoicing of expense is wrong')
+
+    def test_expense_multi_id_analytic_distribution(self):
+        """
+        Test conversion of analytic_distribution dict into account numbers when a hr.expense.sheet containing
+        a hr.expense with an analytic_distribution having 2+ account ids
+        """
+        product = self.env['product.product'].create({
```

Comment:
```suggestion
        expensed_product = self.env['product.product'].create({
```
Just to make it more explicit.
It could also be nice to add a comment explaining why you need to create a different/dedicated product :nerd_face: